src/components/FreeRoomsCardList.tsx

```diff
@@ -11,26 +13,44 @@ import room_ids from "../../data/room_ids.json";
 export const FreeRoomsCardList: FC<{
   searchText: string;
   timeSlot: string;
-  stringDay: string;
-}> = ({ searchText, timeSlot, stringDay }) => {
-  const rooms = Object.keys(room_occupancy[stringDay]).sort();
-
-  const fliteredRooms = rooms.filter((room: string) => {
-    const hasRoomName = room.includes(searchText.toUpperCase());
-    const hasAvailableSlot =
-      timeSlot == "All" ? true : room_occupancy[stringDay][room][timeSlot];
-
-    return hasRoomName && hasAvailableSlot;
-  });
+  currentDay: string;
+}> = ({ searchText, timeSlot, currentDay }) => {
+  const [fliteredRooms, setFliteredRooms] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const URLsearchText = encodeURIComponent(searchText);
```

Comment:
try to put this line just before fetch occurs and

rename `URLsearchText` to `encodedSearchText`